app/src/main/java/org/schabi/newpipe/player/PlayQueueActivity.java

```diff
@@ -217,11 +217,16 @@ public void onServiceDisconnected(final ComponentName name) {
             }
 
             @Override
-            public void onServiceConnected(final ComponentName name, final IBinder service) {
+            public void onServiceConnected(final ComponentName name, final IBinder binder) {
                 Log.d(TAG, "Player service is connected");
 
-                if (service instanceof PlayerService.LocalBinder) {
-                    player = ((PlayerService.LocalBinder) service).getPlayer();
+                if (binder instanceof PlayerService.LocalBinder localBinder) {
+                    final @Nullable PlayerService s = localBinder.getService();
+                    if (s == null) {
+                        player = null;
+                    } else {
+                        player = s.getPlayer();
+                    }
                 }
```

Comment:
```suggestion
                final var service = binder instanceof PlayerService.LocalBinder localBinder ? localBinder.getService() : null;
                player = service != null ? service.getPlayer() : null;
```

Reply:
That’s a lot more complicated to read than without tertiaries!